src/transformers/models/dac/modeling_dac.py

```diff
@@ -520,34 +521,38 @@ def apply_weight_norm(self):
             weight_norm(layer.res_unit3.conv1)
             weight_norm(layer.res_unit3.conv2)
 
-    def remove_weight_norm(self):
+    def remove_weight_norm(self, old_weight_norm=False):
+        remove_weight_norm = nn.utils.remove_weight_norm
+        if hasattr(nn.utils.parametrizations, "weight_norm") and not old_weight_norm:
+            remove_weight_norm = torch.nn.utils.parametrize.remove_parametrizations
```

Comment:
Felt strange that `add_weight_norm` would default to the newer function while `remove_weight_norm` wouldn't. That would lead to error but also add the flag `old_weight_norm` so can use corresponding removal function if needed